src/CommunityToolkit.Maui/PopupService.cs

```diff
@@ -0,0 +1,127 @@
+﻿using CommunityToolkit.Maui.Views;
+using System.ComponentModel;
+
+namespace CommunityToolkit.Maui;
+
+/// <inheritdoc cref="IPopupService"/>
+public class PopupService : IPopupService
+{
+	readonly IServiceProvider serviceProvider;
+
+	static readonly IDictionary<Type, Type> viewModelToViewMappings = new Dictionary<Type, Type>();
```

Comment:
Why have this mapping when the IServiceProvider can handle the view-to-viewmodel mapping for you?  Is the assumption that Popup view implementations won't make use of DI? Or is it more about invoking the popup via the viewmodel type rather than the view type in application code?

Reply:
That is a good question! The initial plan was to make it cleaner to show a Popup (View) from a View Model by only knowing the type of the view model and not the view.

Reply:
It is possible to use the IServiceProvider to that benefit.  This way there is no worry about being in sync with other type mappings that may live in IServiceProvider.

`
internal interface ITypeMapper<TInType>
{
    Type MappedType { get; }
}

internal class ViewModelToViewTypeMapper<TInType, TOutType> : ITypeMapper<TInType>
{
     Type MappedType => typeof(TOutType);
}
`
Usage:
`
_services.AddSingleton<ITypeMapper<TViewModel>, ViewModelToViewTypeMapper<TViewModel, TView>>();
...
var viewType = _serviceProvider.GetRequiredService<ITypeMapper<TViewModel>>().MappedType;
`

Either way, small potatoes.

Reply:
It is possible to use the IServiceProvider to that benefit.  This way there is no worry about being in sync with other type mappings that may live in IServiceProvider.

```
internal interface ITypeMapper<TInType>
{
    Type MappedType { get; }
}

internal class ViewModelToViewTypeMapper<TInType, TOutType> : ITypeMapper<TInType>
{
     Type MappedType => typeof(TOutType);
}
```
usage:
```
_services.AddSingleton<ITypeMapper<TViewModel>, ViewModelToViewTypeMapper<TViewModel, TView>>();
...
var viewType = _serviceProvider.GetRequiredService<ITypeMapper<TViewModel>>().MappedType;
```